Update httputils tests

vdemeester opened the pull request:
This updates a bit the `httputils` tests (again 🐙) and covers a little but more.
- Make sure we use `httptest` instead of depending on network.
- Update `resumablerequestreader` tests.

``` bash
# Before
+ go test -test.timeout=60m github.com/docker/docker/pkg/httputils
PASS
coverage: 65.6% of statements
# After
+ go test -test.timeout=60m github.com/docker/docker/pkg/httputils
time="2015-09-06T13:50:33Z" level=info msg="encountered error during pull and clearing it before resume: A error occured" 
PASS
coverage: 98.4% of statements
```
- There is some duplicated code in tests, I'll look for a way to reduce this a bit :sweat_smile:.
- `resumablerequestreader` takes some times because of the `time.Sleep` present in the code (sleeping for 5 seconds or more) — I was wondering if we could/should make it parametrizable (but it would be mainly for tests..).

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

runcom commented on pkg/httputils/httputils_test.go:
```diff
@@ -1,25 +1,65 @@
+	response.Body.Close()
+
+	if err != nil || string(actual) != expected {
+		t.Fatalf("Expected the response '%s', got err:%v, response:%v, actual:%s", expected, err, response, string(actual))
```
you can use `%q` to quote strings instead of '%s'

runcom commented on pkg/httputils/resumablerequestreader_test.go:
```diff
@@ -9,6 +10,228 @@ import (
+		t.Fatalf("Expected to have read nothing, but read %v", read)
+	}
+}
+func TestResumableRequestReaderWithEOFWith416Response(t *testing.T) {
```
it's missing a space above :heart: ocd

runcom commented:
small nits, LGTM otherwise

vdemeester committed:
Update httputils tests

- Make sure we use httptest instead of depending on network
- Update resumablerequestreader tests

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 commented:
LGTM
